lib/sort-order-selector/index.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { Fragment, FunctionComponent, useEffect, useRef } from 'react';
+import { connect } from 'react-redux';
+import { toggleSortOrder, setSortType } from '../state/settings/actions';
+import { recordEvent } from '../state/analytics/middleware';
+
+import * as S from '../state';
+import * as T from '../types';
+
+type StateProps = {
+  shouldDisplay: boolean;
+  sortReversed: boolean;
+  sortType: T.SortType;
+};
+
+type DispatchProps = {
+  setSortType: (sortType: T.SortType) => any;
+  toggleSortOrder: () => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+export const SortOrderSelector: FunctionComponent<Props> = ({
+  shouldDisplay,
+  setSortType,
+  sortReversed,
+  sortType,
+  toggleSortOrder,
+}) => {
+  const changeSort = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedOption = event.currentTarget.selectedOptions[0];
```

Comment:
did you find `HTMLSelectElement.selectedIndex`?

Reply:
Yes but is dropdown.options[selectedIndex] better in some way than dropdown.selectedOptions[0]?

Reply:
only in the sense that it relies on named things vs. the assumption of 0. I know that I wouldn't expect the selected option to always be the first one in the list, or that's how I read this line. I'm guessing this is a list of all possible selections because a drop down can have multiple selected items?

otherwise we're getting the actual option I think, if using seletedIndex